server/app/src/main/java/com/google/heartrate/wearos/app/gatt/attributes/GattCharacteristic.java

```diff
@@ -0,0 +1,240 @@
+            throw new GattException(e);
+        }
+    }
+
+    /**
+     * Set int value to characteristic.
+     *
+     * @param value  value to set
+     * @param format format at which the value should be set
+     * @param offset offset at which the value should be set
+     * @throws GattException if value can not be set
+     */
+    protected void setIntValue(int value, int format, int offset) throws GattException {
+        if (!mBluetoothGattCharacteristic.setValue(value, format, offset)) {
+            throw new GattException(String.format("Value %d in format %d with offset %d has not been " +
+                    "set to characteristic %s.", value, format, offset, getUUid()));
+        }
+    }
+
+    /**
+     * Set value to characteristic.
+     *
+     * @param value value to set
+     * @throws GattException if value can not be set
+     */
+    protected void setValue(byte[] value) throws GattException {
+        if (!mBluetoothGattCharacteristic.setValue(value)) {
+            throw new GattException(String.format("Value %s can not be " +
+                    "set to characteristic %s.", Arrays.toString(value), getUUid()));
+        }
+    }
+
+    /**
+     * Assert characteristic has read permissions.
+     *
+     * @throws GattException when characteristic has no read permissions
+     */
+    public void assertCharacteristicReadable() throws GattException {
+        if ((mBluetoothGattCharacteristic.getProperties() &
+                (BluetoothGattCharacteristic.PROPERTY_READ)) == 0) {
+            throw new GattException(String.format("Characteristic %s has no read permissions", getUUid()),
+                    BluetoothGatt.GATT_READ_NOT_PERMITTED);
+        }
+    }
+
+    /**
+     * Assert characteristic has write permissions.
+     *
+     * @throws GattException when characteristic has no write permissions
+     */
+    public void assertCharacteristicWritable() throws GattException {
+        if ((mBluetoothGattCharacteristic.getProperties() &
+                (BluetoothGattCharacteristic.PROPERTY_WRITE |
+                        BluetoothGattCharacteristic.PROPERTY_WRITE_NO_RESPONSE)) == 0) {
+            throw new GattException(String.format("Characteristic %s has no write permissions", getUUid()),
+                    BluetoothGatt.GATT_WRITE_NOT_PERMITTED);
+        }
+    }
+
+    /**
+     * Assert characteristic has notify permissions.
+     *
+     * @throws GattException when characteristic has no notify permissions
+     */
+    public void assertCharacteristicNotifiable() throws GattException {
+        if ((mBluetoothGattCharacteristic.getProperties() &
+                BluetoothGattCharacteristic.PROPERTY_NOTIFY) == 0) {
+            throw new GattException(String.format("Characteristic %s has no write permissions", getUUid()),
+                    BluetoothGatt.GATT_REQUEST_NOT_SUPPORTED);
+        }
+    }
+
+    /**
+     * Determine weather characteristic requires write response.
```

Comment:
nit: typo weather -> wether

Reply:
Fixed in [commit](https://github.com/googleinterns/heartrate-bt-wear/pull/1/commits/f81955e6ee20af370c211fd96123abd538b599e1)